From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Danny Lin <danny@kdrag0n.dev>
Date: Tue, 8 Mar 2022 20:37:33 -0800
Subject: [PATCH 17/24] SystemUI: Follow light/dark theme in power menu

Now that we've modified the power menu to refresh on UI mode changes,
make it follow the system light/dark theme for better integration in
light mode.

SystemUI: Always refresh power menu on UI mode change

This is necessary for reliably theming the global actions dialog with
dynamic colors and adapting it to light/dark themes.

Demo screenshots (with color overlays applied):
https://twitter.com/kdrag0n/status/1445960685427433473

[kdrag0n: ported to 12L]

Change-Id: If58fb4079a4cd11414ff928fad3576beecb14ff5
Signed-off-by: althafvly <althafvly@gmail.com>
[a17: the dialog moved to ActionsDialogLiteDelegate (mDelegate); the power menu
 lite colours are drawables that already follow the theme on Android 17, so
 only the UI-mode refresh is carried]

(cherry picked from commit 275a7e6c45d845ca2c7c6e4a600bf958ef32853c)
---
 .../systemui/globalactions/GlobalActionsDialogLite.java  | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/packages/SystemUI/src/com/android/systemui/globalactions/GlobalActionsDialogLite.java b/packages/SystemUI/src/com/android/systemui/globalactions/GlobalActionsDialogLite.java
index 26c5a40aad4a..d0616e6e422c 100644
--- a/packages/SystemUI/src/com/android/systemui/globalactions/GlobalActionsDialogLite.java
+++ b/packages/SystemUI/src/com/android/systemui/globalactions/GlobalActionsDialogLite.java
@@ -1024,6 +1024,15 @@ public class GlobalActionsDialogLite implements DialogInterface.OnDismissListene
                 user.id) != 0;
     }
 
+    @Override
+    public void onUiModeChanged() {
+        // Colors may change, depending on UI mode
+        mContext.getTheme().applyStyle(mContext.getThemeResId(), true);
+        if (mDelegate != null && mDelegate.isShowing()) {
+            mDelegate.refreshDialog();
+        }
+    }
+
     @Override
     public void onConfigChanged(Configuration newConfig) {
         if (mDelegate != null && mDelegate.isShowing()
